Sources/StreamChatUI/ChatMessageList/ChatChannelHeaderView.swift

```diff
@@ -0,0 +1,144 @@
+    open var timer: Timer?
+
+    /// The amount of time it updates the online status of the members.
+    /// By default it is 60 seconds.
+    open var statusUpdateInterval: TimeInterval { 60 }
+
+    /// A view that displays a title label and subtitle in a container stack view.
+    open private(set) lazy var titleContainerView: TitleContainerView = components
+        .titleContainerView.init()
+        .withoutAutoresizingMaskConstraints
+
+    override open func setUp() {
+        super.setUp()
+
+        channelController?.setDelegate(self)
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+
+        embed(titleContainerView)
+    }
+
+    override open func updateContent() {
+        super.updateContent()
+
+        titleContainerView.content = (titleText, subtitleText)
+
+        makeTimer()
+    }
+
+    /// The title text used to render the title label. By default it is the channel name.
+    open var titleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        return components.channelNamer(channel, currentUserId)
+    }
+
+    /// The subtitle text used in the subtitle label. By default it shows member online status.
+    open var subtitleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        guard let currentUserId = self.currentUserId else { return nil }
+
+        if channel.isDirectMessageChannel {
+            guard let member = channel
+                .lastActiveMembers
+                .first(where: { $0.id != currentUserId })
+            else {
+                return nil
+            }
+
+            if member.isOnline {
+                return L10n.Message.Title.online
+            } else if let minutes = member.lastActiveAt
+                .flatMap({ DateComponentsFormatter.minutes.string(from: $0, to: Date()) }) {
+                return L10n.Message.Title.seeMinutesAgo(minutes)
+            } else {
+                return L10n.Message.Title.offline
+            }
+        }
+
+        return L10n.Message.Title.group(channel.memberCount, channel.watcherCount)
+    }
+
+    /// Create the timer to repeatedly update the online status of the members.
+    open func makeTimer() {
+        // Only create the timer if is not created yet and if the interval is not zero.
+        guard timer == nil, statusUpdateInterval > 0 else {
+            return
+        }
+
+        timer = Timer.scheduledTimer(
+            withTimeInterval: statusUpdateInterval,
+            repeats: true
+        ) { [weak self] _ in
+            self?.updateContentIfNeeded()
+        }
+    }
+
+    // MARK: - ChatChannelControllerDelegate Implementation
+
+    open func channelController(
+        _ channelController: _ChatChannelController<ExtraData>,
+        didUpdateChannel channel: EntityChange<_ChatChannel<ExtraData>>
+    ) {
+        switch channel {
+        case .update:
+            updateContentIfNeeded()
```

Comment:
```suggestion
        switch channel {
        case .update, .create:
            updateContentIfNeeded()
```

Reply:
Why do you think "create" is needed ? 🤔 What is the use case?

Reply:
the use case is showing a message list when channel hasn't yet been created (an optimistic update). In that case, we would get `create` change first (most likely followed by several `update` but I wouldn't rely on that)

Reply:
✅